### PR TITLE
[C-2653] Add shallow argument to audius-query hooks

### PR DESCRIPTION
**Pull request by amendelsohn**
### Description

if `shallow: true`, don't rehydrate the full entity structure.

In the case where kind is undefined, the top-level data is not an entity, so just returning the nonNormalizedData will work. However, if `kind: true`, nonNormalizedData just contains an id, so we need to do one level of denormalization. We provide only the matching kind so we don't repopulate the nested entities.

### CODE REVIEW

File: packages/common/src/audius-query/utils.ts [hunk 1]
```diff
@@ -15,8 +16,18 @@ export const getKeyFromFetchArgs = (fetchArgs: any) => {
 
 export const selectCommonEntityMap = (
   state: CommonState,
-  kind?: Kind
+  kind?: Kind,
+  shallow?: boolean
 ): EntityMap | null => {
+  if (kind && shallow) {
```
Comment (dylanjeffers) [inline packages/common/src/audius-query/utils.ts:22]: sick so basically we dont need to grab the whole state right?

**Review by dylanjeffers — APPROVED** [review]
love this addition! extra credit for v2 is to add docs in the form of js-doc, and then can generate readme args (or just have them as dupes :))

File: packages/common/src/audius-query/utils.ts (hunk 1 above)
Comment (amendelsohn) [reply packages/common/src/audius-query/utils.ts:22]: right and we only need to do this if the top level data is itself an entity. Without this step, we'd just get `useGetTrackById({ id: 123 }, { shallow: true }) returns { data: 123 }`

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/audius-query-shallow

**Pull request merged**